src/pip_preserve_requirements/_vcs_git.py

```diff
@@ -29,7 +29,11 @@ def place_tag_on_commit(
     ) -> None:
         with tempfile.TemporaryDirectory() as tmpdir:
             subprocess.run(
-                ["git", "clone", "--bare", "--filter=blob:none", source_repo, tmpdir],
+                ["git", "clone", "--depth=1", source_repo, tmpdir],
```

Comment:
```suggestion
                ["git", "clone", "--bare", "--depth=1", source_repo, tmpdir],
```

Does this work with --bare too?